### PR TITLE
Storage Validation: Promote CSI Volume Attach tests

### PR DESCRIPTION
**Pull request by brahmaroutu**
<!--  Thanks for sending a pull request!  Here are some tips for you:

This PR is part of series of Conformance Tests for Storage CSI work.
/release-note-none
/sig-storage
/assign msau42

### CODE REVIEW

**Comment by dims** [comment]
@brahmaroutu Let's add the tests without marking them as conformance, get them running first, monitor for flakiness etc and then promote them

https://github.com/kubernetes/community/blob/master/contributors/devel/conformance-tests.md#promoting-tests-to-conformance

cc @bgrant0607 @kubernetes/cncf-conformance-wg

File: test/e2e/storage/csi_conformance.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+		if framework.TestContext.ReportDir == "" {
+			podlogs.WatchPods(ctx, cs, ns.Name, GinkgoWriter)
+		}
+	})
+
+	AfterEach(func() {
+		cancel()
+	})
+
+	Context("CSI attach test using HostPath driver [Feature:CSIDriverRegistry]", func() {
```
Comment (msau42) [inline test/e2e/storage/csi_conformance.go]: These test cases already exist so I don't think we need to copy them as new test suites.

**Review by msau42 — COMMENTED** [review]
Thanks @brahmaroutu for starting to take a look at this.  Is there an issue with a testing plan that lists the test cases we are trying to add/promote to conformance?

File: test/e2e/storage/csi_conformance.go (hunk 1 above)
Comment (brahmaroutu) [reply test/e2e/storage/csi_conformance.go]: Conformance tests require to be individual tests and cannot be in table format, each with their own 'It' block without a loop that programmatically creates the tests and each of them will have their own documentation. Unfortunately the framework CSI tests have will not work here. Please advice.

Comment (msau42) [reply test/e2e/storage/csi_conformance.go]: Ok, let's at least try to refactor the tests into a common function that can be shared by conformance vs non-conformance tests. But also, I don't think we should use these CSI attach tests.  They are more like integration tests and do not really test end-user behavior.

I outlined some possible test cases we can add here: https://github.com/kubernetes/kubernetes/issues/69602

**Comment by msau42** [comment]
Hi @brahmaroutu, I still don't think this is a good test case to consider for a storage validation suite for the following reasons:
* It's not testing end-user experience, it's testing implementation details
* It's using a mock driver and not a real volume driver
* It's an alpha feature (soon to be beta)

Instead, can we focus on the test cases that I outlined in https://github.com/kubernetes/kubernetes/issues/69602?

**Comment by brahmaroutu** [comment]
@msau42 we can keep this on hold for now, I will work on other test cases.

**Pull request closed without merging**